src/pages/Vaults.tsx

```diff
@@ -155,7 +181,7 @@ export function Vaults() {
       <PageContent>
         <ValueCardGrid>
           <VaultManagerCountCard totalCollateralTypes={Object.keys(dashboardData)?.length} isLoading={isLoading} />
-          <ActiveVaultCountCard activeVaults={openVaults?.length} isLoading={isLoading} />
+          <ActiveVaultCountCard activeVaults={openVaults?.length} isLoading={isLoading || nextPagesIsLoading} />
           <VaultTotalLockedCollateralValueCard data={dashboardData} isLoading={isLoading} />
```

Comment:
You can `isLoading={isLoading || nextPagesIsLoading} ` for other components too. Ideally storing `isLoading || nextPagesIsLoading` in a variable and passing over to the child components.

Reply:
Our main source of data is `vaultsDataAppended`, Right? Based on this conditional:
```
  if (vaultsDataAppended && Object.keys(vaultsDataAppended).length > 0) {
    [openVaults, dashboardData, tokenNames] = processVaultsData(vaultsDataAppended);
  }
``
We're processing `vaultsDataAppended` to get `openVaults, dashboardData, tokenNames`. And you fetch `vaultsDataAppended` data with a `useSWR` call. Based on that I think, it's best to handle it.